### PR TITLE
Added `is_secret_data_base64` field to `google_secret_manager_secret_version`

### PR DESCRIPTION
**Pull request by abheda-crest**
Added the `is_secret_data_base64` field to the `google_secret_manager_secret_version` resource
fixes https://github.com/hashicorp/terraform-provider-google/issues/10129

**Release Note Template for Downstream PRs (will be copied)**

```release-note:enhancement
secretmanager: added `is_secret_data_base64` field to `google_secret_manager_secret_version` resource
```

### CODE REVIEW

**Comment by abheda-crest** [comment]
Just to clarify, while importing the `secret_version` resources having the `is_secret_data_base64` field set to true, the `secret_data` is decoded from the base64-encoded string received from API. It is potentially due to the [condition of the flattener](https://github.com/GoogleCloudPlatform/magic-modules/pull/8873/files#diff-4f39833f58f47e80260cf3b3ca754b1ad5492f0881c3edc568f60132ca55c203R48). It seems the import is working fine when the `is_secret_data_base64` is not provided or set to false(again due to the same condition).

**Comment by abheda-crest** [comment]
@trodge This PR is now ready for review

File: mmv1/templates/terraform/extra_schema_entry/secret_version_is_secret_data_base64.go.erb
Comment (trodge) [inline mmv1/templates/terraform/extra_schema_entry/secret_version_is_secret_data_base64.go.erb:1]: Why does this field need to be added this way as opposed to being in the yaml properties?

Comment (abheda-crest) [reply mmv1/templates/terraform/extra_schema_entry/secret_version_is_secret_data_base64.go.erb:1]: This field is a terraform-only field that decides whether the `secret_data` is to be sent as is or after base64-encoding and whether to decode it while flattening. As this field affects the state value of `secret_data`, we need to keep it `ForceNew`. The `virtual_field` property doesn't provide a way to mark the fields immutable. Hence, I've used `extra_schema_entry` for this purpose.

**Review by trodge — APPROVED** [review]

**Pull request merged**